content/language/R/containerize.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Containerize a R application
+keywords: R, flask, containerize, initialize
+description: Learn how to containerize a R application.
+aliases:
+  - /language/R/build-images/
+  - /language/R/run-containers/
+---
+
+## Prerequisites
+
+* You have installed the latest version of [Docker Desktop](../../get-docker.md).
+* You have a [git client](https://git-scm.com/downloads). The examples in this section use a command-line based git client, but you can use any client.
+
+## Overview
+
+This section walks you through containerizing and running a R application.
+
+## Get the sample application
+
+The sample application uses the popular [Shiny](https://shiny.posit.co/) framework.
+
+Clone the sample application to use with this guide. Open a terminal, change directory to a directory that you want to work in, and run the following command to clone the repository:
+
+```console
+$ git clone https://github.com/mfranzon/r-docker.git
+```
+
+## Initialize Docker assets
```

Comment:
The sample repository already has a compose file and a Dockerfile, resulting in a warning on `docker init`. Should those files be used, or should `docker init` generate files? 

```text
! Warning → The following Docker files already exist in this directory:
  - Dockerfile
  - compose.yaml
```

*Edit*: doesn't look like `docker init` supports R, so we'd have to use preexisting files (or write them from scratch in the guide)

Reply:
The base image in the existing Dockerfile is a single-platform `linux-amd64` image. To run it on ARM-based systems we'd have to either:

- override the platform `FROM --platform=linux/amd64`
- Build a multi-platform variant of `rocker/shiny`
- Use the `r-base` Docker Official image

Reply:
The `compose.yaml` has a version property. That should be removed.

```yaml
version: '3'
```

Reply:
Thank you for the detailed review! 


> The sample repository already has a compose file and a Dockerfile, resulting in a warning on `docker init`. Should those files be used, or should `docker init` generate files?
> 
> ```
> ! Warning → The following Docker files already exist in this directory:
>   - Dockerfile
>   - compose.yaml
> ```
> 
> _Edit_: doesn't look like `docker init` supports R, so we'd have to use preexisting files (or write them from scratch in the guide)

I have removed `docker init` from all the sections because as you suggests it does not support R so it's better to use the provided files from the repositories.





> The base image in the existing Dockerfile is a single-platform `linux-amd64` image. To run it on ARM-based systems we'd have to either:
> 
>     * override the platform `FROM --platform=linux/amd64`
> 
>     * Build a multi-platform variant of `rocker/shiny`
> 
>     * Use the `r-base` Docker Official image


This is a very good point, I have changed the base images of the dockerfiles to `r-base` as you suggested.




> The `compose.yaml` has a version property. That should be removed.
> 
> ```yaml
> version: '3'
> ```

Done!